x-pack/plugins/fleet/public/applications/fleet/sections/agent_policy/create_package_policy_page/components/steps/components/package_policy_input_var_field.tsx

```diff
@@ -376,12 +376,37 @@ function SecretInputField({
   setIsDirty,
   isDirty,
 }: InputComponentProps) {
-  const [editMode, setEditMode] = useState(isEditPage && !value);
+  const [isReplacing, setIsReplacing] = useState(isEditPage && !value);
   const valueOnFirstRender = useRef(value);
 
+  const hasExistingValue = !!valueOnFirstRender.current;
   const lowercaseTitle = varDef.title?.toLowerCase();
+  const showInactiveReplaceUi = isEditPage && !isReplacing && hasExistingValue;
+  const valueIsSecretRef = value && value?.isSecretRef;
+
+  const inputComponent = getInputComponent({
+    varDef,
+    value: isReplacing && valueIsSecretRef ? '' : value,
+    onChange,
+    frozen,
+    packageName,
+    packageType,
+    datastreams,
+    isEditPage,
+    isInvalid,
+    fieldLabel,
+    fieldTestSelector,
+    isDirty,
+    setIsDirty,
+  });
+
+  // If there's no value for this secret, display the input as its "brand new" creation state
+  // instead of the "replace" state
```

Comment:
would it be easy to add unit tests on this component?

Reply:
It's a little complicated for now but definitely doable, so I filed https://github.com/elastic/kibana/issues/172089 as a tech debt issue to follow up on this.